3.3/.sti/bin/assemble

```diff
@@ -1,23 +1,66 @@
 #!/bin/bash
 
-# For SCL enablement
-source .bashrc
+# This script uses these environment variables:
+#
+# DISABLE_COLLECTSTATIC: if not empty, inhibits execution of 'manage.py collectstatic'.
 
-set -e
+function assemble() {
+  # For SCL enablement
+  source .bashrc
 
-echo "---> Copying application source ..."
-cp -Rf /tmp/src/. ./
+  set -e
 
-if [ -f setup.py ]; then
-    echo "---> Installing application ..."
-    python setup.py install --user
-elif [ -f requirements.txt ]; then
-    echo "---> Installing dependencies ..."
-    pip install --user -r requirements.txt
-fi
+  echo "---> Copying application source ..."
+  cp -Rf /tmp/src/* ./
 
-# set permissions for any installed artifacts
-chmod -R og+rwx /opt/openshift
+  if [ -f requirements.txt ]; then
+      echo "---> Installing dependencies ..."
+      pip install --user -r requirements.txt
+  fi
 
-# remove pip temporary directory
-rm -rf /tmp/pip_build_default
+  if [ -f setup.py ]; then
+      echo "---> Installing application ..."
+      python setup.py install --user
+  fi
+
+  # set permissions for any installed artifacts
+  chmod -R og+rwx /opt/openshift
+
+  set +e
```

Comment:
It wouldn't be right to have it as part  of `django_collectstatic`.
The reasoning here is to avoid side effects, keep them in as few places as possible. `set +e; set -e` change the behavior of the current shell, if I put it inside `django_collectstatic`, after calling that function the caller will inadvertently have the '-e' flag enabled... even if it was not set before.

If I knew how to do it, I'd save the state of the shell flags in the beginning of `django_collectstatic`, do `set +e`, and later set the flags back to the original value.